lib/prom_ex/grafana_client/connection.ex

```diff
@@ -24,6 +24,14 @@ defmodule PromEx.GrafanaClient.Connection do
     }
   end
 
+  def build(finch_process, %{host: host, auth_token: {m, f, a}}) do
+    %__MODULE__{
+      finch_process: finch_process,
+      base_url: normalize_host(host),
+      authorization: "Bearer #{apply(m, f, a)}"
+    }
+  end
+
```

Comment:
Why not accept a function too?

Reply:
My reason is that the mfa is coming in from a config file, and anonymous functions are a no-go there. Probably, an anonymous function could be accepted if passed in another way, but it was out of scope with what I was going for.